docs/generate-time-series/javascript.rst

```diff
@@ -0,0 +1,241 @@
+
+.. code-block:: console
+
+    $ npm install pg
+
+You will also be using the `Axios`_ library to make HTTP queries. Install the ``axios``
+library with npm:
+
+.. code-block:: console
+
+    $ npm install axios
+
+
+Both the ``axios`` and ``node-postgres`` libraries operate using `Promises`_. Promises are
+a way of encapsulating the eventual result of an asynchronous operation. Both ``axios``, when
+retrieving content from a URL, and ``node-postgres``, when connecting and querying a database,
+use Promises as a way of asynchronously doing those operations.
+
+If you are not familiar with asynchronous operations and Promises, Mozilla have written a
+`detailed guide`_ on the subject.
+
+Most of this tutorial is designed for Node's `interactive REPL mode`_ so that
+you can experiment with the commands as you see fit. Since both the libraries you will import
+use Promises, you should start ``node`` with support for the `await`_ operator:
+
+.. code-block:: console
+
+    $ node --experimental-repl-await
+
+
+Use Javascript to get to get ISS telemetry data
+===============================================
+
+You will be consuming telemetry data from `Open Notify`_, a service which provides
+a simple API to consume data from NASA. One of these data points is the current location
+of the International Space Station. The endpoint for these data can be found at
+`<http://api.open-notify.org/iss-now.json>`_.
+
+Start an interactive Node session using the ``node`` command. You can read the current
+position of the ISS by consuming this endpoint, using ``axios``::
+
+    > const axios = require('axios').default;
+    > let response = await axios.get('http://api.open-notify.org/iss-now.json')
+    > response.data
+    { iss_position: { longitude: '111.5276', latitude: '9.7319' },
+    timestamp: 1582101199,
+    message: 'success' }
+
+The endpoint returns a JSON payload consisting of ``iss_position``, which is composed of
+``latitude`` and ``longitude``, as well as some other metadata.
+
+You can encapsulate this within a single function to return only the longitude and latitude as a `WKT`_
+string::
+
+    > async function position() {
+    ...     let response = await axios.get('http://api.open-notify.org/iss-now.json')
+    ...     return `POINT (${response.data.iss_position.longitude} ${response.data.iss_position.latitude})`
+    ... }
+
+When you run this function, it should return your point string::
+
+    > await position()
+    'POINT (-22.7857 49.3332)'
+
+Set up CrateDB
+==============
+
+First, import the `pg`_ client::
+
+    > const { Client } = require('pg')
+
+Then `connect`_ to CrateDB, using the port for the `Postgres Wire Protocol`_::
+
+    > const client = new Client({connectionString: 'postgresql://crate@localhost:5432/doc'})
+    > await client.connect()
+
+Then finally `create a table`_ suitable for writing ISS position coordinates::
+
+    > var query = `
+    ...     CREATE TABLE iss_position (
```

Comment:
perhaps we should use "iss" instead of "iss_position". keeps things simple, and allows for us to maybe expand on what data we include in the future